src/atomate2/classical_md/openmm/jobs/base.py

```diff
@@ -0,0 +1,494 @@
+                )
+
+            traj_file = dir_name / f"{traj_file_name}.{traj_file_type}"
+
+            if traj_file.exists() and task_reports(prev_task, "traj"):
+                self.traj_file_name = increment_name(traj_file_name)
+
+            # TODO: MDA 2.7.0 has a bug that prevents velocity reporting
+            #  this is a stop gap measure before MDA 2.8.0 is released
+            kwargs = dict(
+                file=str(dir_name / f"{self.traj_file_name}.{traj_file_type}"),
+                reportInterval=traj_interval,
+                enforcePeriodicBox=self._resolve_attr("wrap_traj", prev_task),
+            )
+            if report_velocities:
+                # assert package version
+
+                kwargs["writer_kwargs"] = writer_kwargs
+                warnings.warn(
+                    "Reporting velocities is only supported with the"
+                    "development version of MDAnalysis, >= 2.8.0, "
+                    "proceed with caution.",
+                    stacklevel=1,
+                )
+            traj_reporter = MDAReporter(**kwargs)
+
+            sim.reporters.append(traj_reporter)
+
+        # add state reporter
+        state_interval = self._resolve_attr("state_interval", prev_task)
+        state_file_name = self._resolve_attr("state_file_name", prev_task)
+        if has_steps & (state_interval > 0):
+            state_file = dir_name / f"{state_file_name}.csv"
+            if state_file.exists() and task_reports(prev_task, "state"):
+                self.state_file_name = increment_name(state_file_name)
+
+            state_reporter = StateDataReporter(
+                file=f"{dir_name / self.state_file_name}.csv",
+                reportInterval=state_interval,
+                step=True,
+                potentialEnergy=True,
+                kineticEnergy=True,
+                totalEnergy=True,
+                temperature=True,
+                volume=True,
+                density=True,
+            )
+            sim.reporters.append(state_reporter)
+
+    def run_openmm(self, simulation: Simulation) -> NoReturn:
+        """Abstract method for running the OpenMM simulation.
+
+        This method should be implemented by subclasses to
+        define the specific simulation logic. It takes an
+        OpenMM simulation object and evolves the simulation.
+
+        Parameters
+        ----------
+        simulation : Simulation
+            The OpenMM simulation object.
+
+        Raises
+        ------
+        NotImplementedError
+            If the method is not implemented by a subclass.
+        """
+        raise NotImplementedError(
+            "`run_openmm` should be implemented by each child class."
+        )
+
+    def _resolve_attr(
+        self,
+        attr: str,
+        prev_task: OpenMMTaskDocument | None = None,
+        add_defaults: dict | None = None,
+    ) -> Any:
+        """Resolve an attribute and set its value.
```

Comment:
I think this would be even simpler if the defaults are on the base class, as then there are only two places to check.

Reply:
See my response above RE:

> What is the benefit of having these here rather than settings them in the BaseOpenMMMaker directly? It seems like this would be cleaner?